### PR TITLE
Add whitelisting of endpoints

### PR DESCRIPTION
**Pull request by AdityaVallabh**
Provide API endpoints to be tested in a `whitelist.txt` and only their test suites will be generated

### CODE REVIEW

File: mqgo/src/meqa/mqgen/main.go [hunk 1]
```diff
@@ -29,19 +29,34 @@ func main() {
...
 	swaggerJsonPath := *swaggerFile
 	if fi, err := os.Stat(swaggerJsonPath); os.IsNotExist(err) || fi.Mode().IsDir() {
 		fmt.Printf("Can't load swagger file at the following location %s", swaggerJsonPath)
 		os.Exit(1)
 	}
+	whitelistPath := *whitelistFile
+	var whitelist map[string]bool
+	if len(whitelistPath) > 0 {
+		if fi, err := os.Stat(whitelistPath); os.IsNotExist(err) || fi.Mode().IsDir() {
+			fmt.Printf("Can't load whitelist file at the following location %s", whitelistPath)
```
Comment (nitika18) [inline mqgo/src/meqa/mqgen/main.go:50]: Here we have used fmt.Printf and in line 56 we are using mqutil.Logger.Printf, any reason for using two different approaches for printing messages.

File: mqgo/src/meqa/mqgen/main.go [hunk 2]
```diff
@@ -29,19 +29,34 @@ func main() {
...
+			os.Exit(1)
+		}
+		wl, err := mqswag.GetWhitelistSuites(whitelistPath)
+		whitelist = wl
+		if err != nil {
+			mqutil.Logger.Printf("Error: %s", err.Error())
```
Comment (nitika18) [inline mqgo/src/meqa/mqgen/main.go:56]: Let's also add a custom message.

File: mqgo/src/meqa/mqswag/parser.go [hunk 3]
```diff
@@ -167,6 +167,20 @@ func CreateSwaggerFromURL(path string, meqaPath string) (*Swagger, error) {
 	return (*Swagger)(specDoc.Spec()), nil
 }
 
+func GetWhitelistSuites(path string) (map[string]bool, error) {
+	whitelistBytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		mqutil.Logger.Printf("can't read file %s", path)
```
Comment (nitika18) [inline mqgo/src/meqa/mqswag/parser.go:173]: So we are printing the error here as well and in the main method as well. We can do it a single place.

File: mqgo/src/meqa/mqgen/main.go [hunk 4]
```diff
@@ -29,19 +29,34 @@ func main() {
...
-func run(meqaPath *string, swaggerFile *string, algorithm *string, verbose *bool) {
+func run(meqaPath *string, swaggerFile *string, algorithm *string, verbose *bool, whitelistFile *string) {
 	mqutil.Verbose = *verbose
 
 	swaggerJsonPath := *swaggerFile
 	if fi, err := os.Stat(swaggerJsonPath); os.IsNotExist(err) || fi.Mode().IsDir() {
 		fmt.Printf("Can't load swagger file at the following location %s", swaggerJsonPath)
 		os.Exit(1)
 	}
+	whitelistPath := *whitelistFile
```
Comment (nitika18) [inline mqgo/src/meqa/mqgen/main.go:46]: Why do we need a different assignment?
We are directly using the value without any changes.

**Comment by AdityaVallabh** [comment]
@nitika18 since this PR is obsolete, making the changes in #22

File: mqgo/src/meqa/mqgen/main.go (hunk 1 above)
Comment (AdityaVallabh) [reply mqgo/src/meqa/mqgen/main.go:50]: `fmt.Print` prints to the `stdout` whereas `mqutil.Logger` can print both to `stdout` as well as a specified log file.
However in this case, we're only using the terminal so logger isn't required.
Will remove it.

**Pull request closed without merging**